include/aws/common/bigint.h

```diff
@@ -0,0 +1,99 @@
+#ifndef AWS_COMMON_BIGINT_H
+#define AWS_COMMON_BIGINT_H
+
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/common.h>
+
+#include <aws/common/array_list.h>
+#include <aws/common/byte_buf.h>
+
+/*
+ * A basic big integer implementation using 2^32 as the base.  Algorithms used are formalizations of the basic
+ * grade school operations everyone knows and loves (as formalized in AoCP Vol 2, 4.3.1).  Current use case
+ * targets do not yet involve a domain large enough that its worth exploring more complex algorithms.
+ */
+struct aws_bigint {
+    /*
+     * A sequence of base 2^32 digits starting from the least significant
+     */
+    struct aws_array_list digits;
+
+    /*
+     * 1 = positive, -1 = negative
+     */
+    int sign;
+};
+
+AWS_EXTERN_C_BEGIN
+
+AWS_COMMON_API
+void aws_bigint_clean_up(struct aws_bigint *bigint);
+
+AWS_COMMON_API
+int aws_bigint_init(struct aws_bigint *bigint, struct aws_allocator *allocator);
+
+AWS_COMMON_API
+int aws_bigint_init_reserve(struct aws_bigint *bigint, struct aws_allocator *allocator, uint64_t reserve_bits);
+
+AWS_COMMON_API
+int aws_bigint_init_from_hex(
+    struct aws_bigint *bigint,
+    struct aws_allocator *allocator,
+    struct aws_byte_cursor hex_digits);
```

Comment:
Nice job accepting input that starts with "0x" but not requiring it.
Add a comment here explaining that.